pkg/kclient/deployments.go

```diff
@@ -42,6 +44,20 @@ func getDeploymentCondition(status appsv1.DeploymentStatus, condType appsv1.Depl
 	return nil
 }
 
+// ListDeployments lists all deployments by application name
+func (c *Client) ListDeployments(appName string) (*appsv1.DeploymentList, error) {
```

Comment:
Can we make this more generic? instead of just using `appName`, we can pass in what labelSelector in string, so it can be used by anyone in the future. 

No unit tests either.

Reply:
have recorded it in the follow up issue